packages/report-server/src/reportBuilder/reportBuilder.ts

```diff
@@ -51,11 +51,19 @@ export class ReportBuilder {
     const builtQuery = await new QueryBuilder(this.reqContext, this.config, query).build();
     const data = await fetch(aggregator, builtQuery);
 
-    const context = await buildContext(this.config.transform, this.reqContext, data);
-    const transform = buildTransform(this.config.transform, context);
+    const transformContext = await buildTransformContext(
+      this.config.transform,
+      this.reqContext,
+      data,
+    );
+    const transform = buildTransform(this.config.transform, transformContext);
     const transformedData = transform(data.results);
 
-    const output = buildOutput(this.config.output);
+    const outputContext = await buildOutputContext(
+      { ...this.config.output, ...this.config.fetch },
+      aggregator,
+    );
+    const output = buildOutput({ ...this.config.output, ...this.config.fetch }, outputContext);
```

Comment:
I find this line a little confusing, why are we combining the fetch and output params here?
```suggestion
    const output = buildOutput(this.config.output, outputContext);
```